src/utils/testing.js

```diff
@@ -66,4 +81,77 @@ export const defaultObjectMetadata = async () => {
   };
 };
 
+export const addBrand = () => new ArticleBrand(testData.brands.default).save();
+
+// TODO: this method could benefit from refactoring. To consider.
+export const addArticles = async (articleBrandId, numberPublished, numberUnpublished) => {
+  const defaultMetadata = await defaultObjectMetadata();
+  const totalNumber = numberPublished + numberUnpublished;
+
+  const articles = [];
+
+  await Promise.all(
+    [...Array(totalNumber).keys()].map(i => {
+      const day = i < 9 ? `0${i + 1}` : i + 1;
+      const passedDate = new Date(`2017-11-${day}T17:25:43.511Z`);
+      const futureDate = new Date(`2027-11-${day}T16:25:43.511Z`);
+      const date = i < numberPublished ? passedDate : futureDate;
+
+      return new Article({
+        brand: articleBrandId,
+        type: 'text',
+        imagePreviewUrl: 'image-url',
+        metadata: defaultMetadata,
+        publishAt: date,
+      })
+        .save()
+        .then(({ _id }) => {
+          articles.push({ _id, publishAt: date, locales: {} });
+        });
+    })
+  );
+
+  articles.sort((a, b) => {
+    if (a.publishAt < b.publishAt) {
+      return -1;
+    }
+    if (a.publishAt > b.publishAt) {
+      return 1;
+    }
+    return 0;
+  });
+
+  await Promise.all(
+    [...Array(totalNumber).keys()].reduce(
```

Comment:
as far as I understand `reduce` is redundant, just use `map`

Reply:
oh, I see, check out [`lodash/flatten`](https://lodash.com/docs/4.17.11#flatten)